### ISSUE TITLE
Provide a non-codegen mockito release with updated test_api dependency.

### ISSUE THREAD

**Issue by jonahwilliams** [dart-lang/mockito#352]
Currently the flutter_tools project and several dev/ bits of the flutter SDK repo depend on mockito. Because we do not want to introduce a codegen requirement for developing with the SDK (and because we'd like to reduce mocks anyway) we have not been able to upgrade past 4.1.1. We anticipate this migrating will take at least a few more months.

Unfortunately mockito 4.1.1 depends on test_api, which means it constrains our test package version and thus everything that depends on test (shelf, shelf_packages, vm_service, dwds, analyzer).

To avoid being stuck at our current version for the next few months, it would be great to have a release of mockito that supported the latest versions but did not require codegen. Its not necessary for this to be null safe.

SDK version <2.12.0 / test_api ^0.3.0

**Comment by srawlins** [comment]
If it's possible (a la constraint solving), should not be hard.

**Comment by srawlins** [comment]
Well... hmm...


```none
$ git diff
diff --git a/pubspec.yaml b/pubspec.yaml
index b291b6d..21e286c 100644
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -15,7 +15,7 @@ dependencies:
   collection: ^1.1.0
   matcher: ^0.12.3
   meta: ^1.0.4
-  test_api: ^0.2.1
+  test_api: ^0.3.0
 
 dev_dependencies:
   build_runner: ^1.0.0
$ pub get
Resolving dependencies... (1.0s)
The current Dart SDK version is 2.10.4.

Because mockito depends on test_api >=0.2.19-nullsafety.5 which requires SDK
version >=2.12.0-0 <3.0.0, version solving failed.
```

So does this 4.1.1+1 also need to be null-safe?

**Comment by srawlins** [comment]
Ah, I just need to run `pub get` with Dart 2.12.

I think I am blocked on getting a build_runner with https://github.com/dart-lang/build/pull/3028 publishd.

```none
Because build_runner >=0.10.1 depends on web_socket_channel ^1.0.9 and test >=1.16.6 depends on web_socket_channel ^2.0.0, build_runner >=0.10.1 is incompatible with test >=1.16.6.
And because test >=1.16.0-nullsafety.19 <1.16.6 depends on test_api 0.2.19 and mockito depends on test_api ^0.3.0, build_runner >=0.10.1 is incompatible with test >=1.16.0-nullsafety.19.
So, because mockito depends on both build_runner ^1.11.2 and test ^1.16.0, version solving failed.
```

**Comment by jonahwilliams** [comment]
Works for me!

**Comment by srawlins** [comment]
Published as 4.1.1+1.
